h/views/activity.py

```diff
@@ -24,6 +24,10 @@
 PAGE_SIZE = 200
 
 
+def _format_date(date):
+    return "{d:%B} {d.day}, {d:%Y}".format(d=date)
```

Comment:
I've never seen this way of formatting dates in Python! Very interesting. So `{d:%B}` produces numbers like `28` and `1` (as opposed to `01`)? And there's no way to do that with `strftime()`?

Anyway, I think you should mover `_format_date()` into a utility module, maybe [h.util.datetime](https://github.com/hypothesis/h/blob/master/h/util/datetime.py), and add some unit tests to it. There's probably not _that_ much to test - but at least test that it works with both double- and single-digit days, and with both date and datetime objects. If nothing else this will prevent someone from coming along and replacing with with a more typical usage of `strftime()` later, and not realising what they're breaking.

Reply:
Heh, I wasn't aware of this either. Any pros/cons compared to `strftime` we should be aware of?

Reply:
I added a comment about this in the commit message. We can't use strftime to produce a non zero padded day as we would use %-d and that isn't supported on all systems. Note strftime calls into the c library of the os and %-d is a 1999 standard where as apparently our qa system (not sure about prod) is 1998 standard and doesn't support it.

Reply:
Thanks, that's interesting. Further discussion: https://hypothes-is.slack.com/archives/C4K6M7P5E/p1519755955000065

I verified that `%-d%` indeed doesn't work in an h Docker container, so this is more portable.